src/main/webapp/courseLoader.js

```diff
@@ -1,164 +1,205 @@
-google.charts.load("current", { packages: ["corechart"] });
+(() => {
+  google.charts.load("current", { packages: ["corechart"] });
+  google.charts.load("current", { packages: ["bar"] });
 
-function fillTitles() {
+  let termData;
```

Comment:
I still don't think you need this to be script-wide. The only places you use it you either already pass in the data or could easily pass it in. 

Up to you if you want to change it. I've just had a lot of nightmares with tracking state like this and I prefer stateless code whenever possible.